backend/routes.go

```diff
@@ -93,18 +98,34 @@ func (s *server) handleGetArticles() http.HandlerFunc {
 
 		var articles = make([]ArticleJson, 0)
 		var article ArticleJson
-		for _, a := range results {
+
+		
+		//Get number of articles necessary
+		//ASSUMPTION: count <= SIZE(RESULTS); will fix this later
+		numArticles, err := strconv.Atoi(mux.Vars(req)["count"])
+		if numArticles == 0 { //NO COUNT SPECIFIED
+			fmt.Println("Getting ALL Articles...")
+			numArticles = len(results)
+		}else{
+			fmt.Printf("Limiting number of articles to  %d \n", numArticles)
+		}
+
+		for _, a := range results[:numArticles] {
```

Comment:
We spoke about it in slack - let's try to do the counting in the database instead of loading all the values into memory first!